### PR TITLE
maxWidth property added for Dialog and DialogTheme 

### LINKED ISSUE #44570: Dialog maxWidth is not configurable
The `Dialog` class uses a `ConstrainedBox` widget with `constraints: const BoxConstraints(minWidth: 280.0)`.

It would be useful to add `maxWidth` as a constructor parameter since currently all dialogs appear very wide on large screens (e.g. tablet or web).

### PR DESCRIPTION
**Pull request by Amitpatil215**
## Description

```double? maxWidth``` property added for ```SimpleDialog```, ```AlertDialog``` and ```Dialog``` widgets and also for ```dialogTheme```.

## Related Issues

fixes #44570

## Tests

I added the following tests:

Verifies ```SimpleDialog``` and ```AlertDialog``` custom ```maxWidth``` property constraints (dialog_test.dart). 
Verifies ```DialogTheme``` ```maxWidth``` property working properly (dialog_theme_test.dart).

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by Piinks** [comment]
Hi @Amitpatil215 I just came across this PR that seems to have aged a bit. 😜 
There are some merge conflicts here, do you want to update and pursue this change?

**Comment by Amitpatil215** [comment]
Thanks for pointing out, Will update the PR!

**Comment by Amitpatil215** [comment]
Hi, @Piinks I have updated PR, Please have a look whenever you get some time. Thanks!

File: packages/flutter/lib/src/material/dialog.dart [hunk 1]
```diff
@@ -140,7 +152,10 @@ class Dialog extends StatelessWidget {
         context: context,
         child: Center(
           child: ConstrainedBox(
-            constraints: const BoxConstraints(minWidth: 280.0),
+            constraints: BoxConstraints(
+              minWidth: 280.0,
+              maxWidth: maxWidth ?? dialogTheme.maxWidth ?? double.infinity,
+              ),
```
Comment (Piinks) [inline packages/flutter/lib/src/material/dialog.dart:158]:
```suggestion
            ),
```

**Review by Piinks — COMMENTED** [review]
I am curious, could the Dialog width not be controlled by the size of the contents? Using your example from the linked issue, if I put the Text in a Container or SizedBox, I can change the width of the Dialog.

**Comment by Amitpatil215** [comment]
> if I put the Text in a Container or SizedBox, I can change the width of the Dialog.

Promising workaround 👍 On behalf of this we can close this PR. Thanks! 

If you think we still need to add this property, comment it down I'm happy to update this PR accordingly!

**Comment by vase4kin** [comment]
Hi, everyone. That would be nice if you still could get it merged. 
My use case is to make the dialog width different based on a device type (adaptive UI). I'd like to control the maximum dialog width based on the theme, not on its content. 
I hope you'll reconsider and we can see it get merged.

**Comment by timbell** [comment]
For those wanting maxWidth functionality you can get it by wrapping the `Dialog` in a `ConstrainedBox` as follows:

```
  Align(
    child: ConstrainedBox(
      constraints: const BoxConstraints(maxWidth: 600),
      child: Dialog(...),
    ),
  )
```

**Pull request closed without merging**